File: proxy/models/book.py
from typing import Optional, Dict, List
from dataclasses import dataclass
from proxy.models.base import Images, Author


@dataclass
class Book:
    id: str
    title: str
    content_type: str
    authors: Optional[List[Author]] = None
    image_url: Optional[str] = None
    release_date: Optional[str] = None
    pages: Optional[int] = None
    description: Optional[str] = None
    images: Optional[Images] = None

    def to_dict(self, images_size: int = 18) -> Dict:
        result = {
            'id': self.id,
            'type': self.content_type,
            'title': self.title,
            'image_url': self.image_url,
            'release_date': self.release_date,
            'pages': self.pages,
            'description': self.description,
        }

        if self.authors:
            result['authors'] = [author.to_dict() for author in self.authors]

        if self.images:
            result['images'] = self.images.to_dict(images_size=images_size)

        return result

File: internal/models/book.go
package models

type Book struct {
	ID          string   `json:"id"`
	Title       string   `json:"title"`
	ContentType string   `json:"type"`
	Authors     []Author `json:"authors,omitempty"`
	ImageURL    *string  `json:"image_url"`
	ReleaseDate *string  `json:"release_date"`
	Pages       *int     `json:"pages"`
	Description *string  `json:"description"`
	Images      *Images  `json:"-"`
}

type BookResponse struct {
	ID          string       `json:"id"`
	Type        string       `json:"type"`
	Title       string       `json:"title"`
	ImageURL    *string      `json:"image_url"`
	ReleaseDate *string      `json:"release_date"`
	Pages       *int         `json:"pages"`
	Description *string      `json:"description"`
	Authors     []Author     `json:"authors,omitempty"`
	Images      []ImageEntry `json:"images,omitempty"`
}

func (b *Book) ToResponse(imagesSize int) BookResponse {
	resp := BookResponse{
		ID:          b.ID,
		Type:        b.ContentType,
		Title:       b.Title,
		ImageURL:    b.ImageURL,
		ReleaseDate: b.ReleaseDate,
		Pages:       b.Pages,
		Description: b.Description,
		Authors:     b.Authors,
	}

	if b.Images != nil {
		resp.Images = b.Images.ToList(imagesSize)
	}

	return resp
}